libs/dyn/convert/to_typed.go

```diff
@@ -246,6 +246,17 @@ func toTypedInt(dst reflect.Value, src dyn.Value) error {
 	case dyn.KindInt:
 		dst.SetInt(src.MustInt())
 		return nil
+	case dyn.KindFloat:
+		v := src.MustFloat()
+		if canConvertToInt(v) {
+			dst.SetInt(int64(src.MustFloat()))
```

Comment:
Can you check what happens if the underlying type is int32 and we use an overflowing integer here?

Reply:
If it overflows it gets set to -1

https://go.dev/play/p/BfJU20dywfp

```
i1: -1
i2: 9223372036854775807
```